android-appcoins-billing/src/main/java/com/appcoins/sdk/billing/helpers/AppcoinsBillingStubHelper.java

```diff
@@ -211,6 +200,26 @@ private String map(SkuDetails skuDetails) {
     }
   }
 
+  private synchronized Bundle walletInstalledBehaviour() throws InterruptedException {
+    isMainThread = Looper.myLooper() == Looper.getMainLooper();
+
+    Bundle response = new Bundle();
+    if (!isServiceBound) {
+      createRepository();
+      synchronized (lockThread) {
+        if (!isMainThread) {
+          lockThread.wait(5000);
```

Comment:
Why was the logic changed to wait for 5 seconds?

Reply:
Not letting the thread wait forever in a scenario of there isn't a connection available.If there is a connection available the thread will be unlock by notify and if it not it will unlock after 5 seconds sending a Interrupt Exception that is handling accordingly.